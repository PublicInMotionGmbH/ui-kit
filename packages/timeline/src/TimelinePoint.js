import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'

const moduleName = 'timeline-point'

const propTypes = {
  /** Additional class name */
  className: PropTypes.string,

  /** Chidren inside Timeline */
  children: PropTypes.node,

  /** Line has special styles */
  special: PropTypes.bool
}

/**
 * Component which represents Timeline.
 *
 * @param {object} props
 * @param {string} [props.className]
 * @param {node} [props.children]
 * @param {bool} [props.special]
 * @returns {React.Element}
 */
function TimelinePoint (props) {
  const { children, special, className } = props

  const clsName = buildClassName(moduleName, className, { special })

  return (
    <div className={clsName}>
      {children}
    </div>
  )
}

TimelinePoint.displayName = 'TimelinePoint'

TimelinePoint.propTypes = propTypes

export default TimelinePoint
