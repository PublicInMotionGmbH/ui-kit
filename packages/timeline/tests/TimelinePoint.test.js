import React from 'react'
import { shallow } from 'enzyme'

import TimelinePoint from '../src/TimelinePoint'

const moduleName = 'talixo-timeline-point'

describe('<TimelineLine />', () => {
  it('renders children correctly', () => {
    const wrapper = shallow(<TimelinePoint />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders class name correctly', () => {
    const wrapper = shallow(<TimelinePoint />)

    expect(wrapper.prop('className')).toBe(moduleName)
  })
})
