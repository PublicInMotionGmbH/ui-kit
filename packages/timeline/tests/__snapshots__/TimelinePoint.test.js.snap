// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelineLine /> renders children correctly 1`] = `
<div
  className="talixo-timeline-point"
/>
`;
